### ISSUE TITLE
Method to get k smallest elements from a tensor

### ISSUE THREAD

**Issue by digital-idiot** [tensorflow/tensorflow#43977]
[`tf.math.top_k`](https://www.tensorflow.org/api_docs/python/tf/math/top_k) only returns `k` largest elements from the input `tensor` along a specified `axis`. However, there is no direct way to obtain `k` smallest elements from a `tensor`. The equivalent method [`topk`](https://pytorch.org/docs/stable/generated/torch.topk.html) in `pytorch` has an `bool` flag `largest` to toggle the output from `k` largest elements to `k` smallest elements and vice versa. It would be useful to have similar option  (or an alternative method like `bottom_k`) for this purpose so that hacky solution like the following can be avoided.

```python
tf.negative(tf.math.top_k(tf.negative(A)))
```
Finding `k` minimum values can be potentially very useful in many cases especially for writing custom `ops`. This does not seem to be too hard to implement or drastically change the API.

**System information**
- TensorFlow version: 2.3

Related discussion in [Stackoverflow](https://stackoverflow.com/questions/44548227/minimum-k-values-of-a-tensor).

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
